medusa/search/core.py

```diff
@@ -951,16 +960,18 @@ def combine_results(multi_results, single_results):
             log.debug(u'All of these episodes were covered by single-episode results,'
                       u' ignoring this multi-episode result')
             continue
+        else:
+            return_multi_results.append(multi_result)
 
         # remove the single result if we're going to get it with a multi-result
         for ep_obj in multi_result.episodes:
-            for i, result in enumerate(single_results):
+            for i, result in enumerate(return_single_results):
                 if ep_obj in result.episodes:
                     log.debug(
                         u'A needed multi-episode result overlaps with a single-episode result'
                         u' for episode {0}, removing the single-episode results from the list',
                         ep_obj.episode,
                     )
-                    del single_results[i]
+                    del return_single_results[i]
 
-    return single_results + result_candidates
+    return return_single_results + return_multi_results
```

Comment:
If this method does its work. A result for a specific shows episode, should only be in `return_single_results` *or* in `return_multi_results` but never in both. Which was the case now.